core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -213,7 +213,8 @@ protected FilterProjectTransposeRule(
   public interface Config extends RelRule.Config {
     Config DEFAULT = ImmutableFilterProjectTransposeRule.Config.of()
         .withOperandFor(Filter.class,
-            f -> !RexUtil.containsCorrelation(f.getCondition()),
+            f -> !RexUtil.containsCorrelation(f.getCondition())
+                && !RexUtil.SubQueryFinder.containsSubQuery(f),
```

Comment:
What about uncorrelated sub-query?

Reply:
Currently, The rule can't push down uncorrelated sub-queries either, because this can be achieved by removing the sub-query through other plans. Even if the sub-query is not removed, the subsequent plans can't be pushed down and will still be fine. 
Correlated sub-queries definitely cannot be pushed down. If pushed down, problems will occur in the subsequent plans. Therefore, here I did not differentiate whether the specific sub-queries contain correlation or not.

Reply:
Shouldn't we be providing a comprehensive function that handles both and use it here then?

Reply:
@asolimando @zabetak Based on the above discussion, I have enhanced RexUtil.containsCorrelation. If the test goes well, I will update the PR again. Maybe we can only push down the non-correlated sub-queries.

Reply:
Thanks @NobiGo, I feel like it's the right approach since all the callers of that function will benefit from this enhancement in a similar way.